app/Libraries/Database.php

```diff
@@ -72,84 +77,212 @@ public static function getInstance(array $credentials = null): object
-          $type = PDO::PARAM_BOOL;
-          break;
-        case is_null($value):
-          $type = PDO::PARAM_NULL;
-          break;
-        default:
-          $type = PDO::PARAM_STR;
-      }
+    $db = Database::getInstance();
+
+    if (!Request::input('db_name')) {
+      return null;
     }
 
-    $this->stmt->bindValue($param, $value, $type);
+    try {
+      $tables = $db->getTablesFromServer(Request::input('db_name'));
+
+      $tables = array_map(function ($table) {
+        $table = array_values($table);
+        return array_pop($table);
+      }, $tables);
+
+      return $tables;
+    } catch (\Throwable $th) {
+      throw $th;
+    }
   }
 
   /**
-   * Execute the prepared statement
+   * Retrieve all Character Sets
    *
-   * @return bool
+   * @return array
    */
-  public function execute(): bool
+  public function getCharsets(): array
   {
-    return $this->stmt->execute();
+    $stmt = self::$pdo->query("SHOW CHARSET");
+    $stmt->execute();
+    return $stmt->fetchAll(PDO::FETCH_ASSOC);
   }
 
   /**
-   * Get result set as array of objects
+   * Retrieve all collations
    *
-   * @return bool
+   * @return array
    */
-  public function resultSet(): bool
+  public function getCollations(): array
   {
-    $this->execute();
-    return $this->stmt->fetchAll(PDO::FETCH_OBJ);
+    $stmt = self::$pdo->query("SHOW COLLATION;");
+    $stmt->execute();
+    return $stmt->fetchAll(PDO::FETCH_ASSOC);
   }
 
   /**
-   * Get single record as object
+   * Create new database on server
    *
-   * @return bool
+   * @param string $dbName
+   * @param string $charset
+   * @param string $collation
+   * @return void
    */
-  public function single(): bool
+  public function createDatabase(string $dbName, string $charset, string $collation): void
   {
-    $this->execute();
-    return $this->stmt->fetch(PDO::FETCH_OBJ);
+    $sql = "CREATE DATABASE $dbName CHARACTER SET $charset COLLATE $collation";
+
+    try {
+      self::$pdo->query($sql);
+    } catch (\Throwable $th) {
+      throw $th;
+    }
   }
 
   /**
-   * Get row count
+   * Retrieves table data
    *
-   * @return int
    */
-  public function rowCount(): int
+  public function fetchTableContents($database, $tableName)
+  {
+    $stmt = self::$pdo->query("USE $database;");
+    $stmt->execute();
+
+    try {
+      $stmt = self::$pdo->query("SELECT * FROM $tableName;");
+      return $stmt->fetchAll(PDO::FETCH_ASSOC);
+    } catch (\Throwable $th) {
+      throw $th;
+    }
+  }
+
+  public function sql($database, $sql)
   {
-    return $this->stmt->rowCount();
+    $stmt = self::$pdo->query("USE $database;");
+    $stmt->execute();
+
+    try {
+      $stmt = self::$pdo->query($sql);
+      return $stmt->fetchAll(PDO::FETCH_ASSOC);
+    } catch (\Throwable $th) {
+      throw $th;
+    }
   }
+
+
+  // /**
+  //  * Prepare statement with query
+  //  *
+  //  * @param string $sql
+  //  * @return void
+  //  */
+  // public function query($sql): void

```

Comment:
If you don't need the commented code, you can always delete it and come back based on the file history to retrieve it if you need it.